### PR TITLE
Add `MiscellaneousSelectVerifier` for `ReportBody` verification

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/1" target="_blank">Graphite</a> 
    * **PR #2** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/2" target="_blank">Graphite</a> 
      * **PR #3** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/3" target="_blank">Graphite</a> 
        * **PR #9** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/9" target="_blank">Graphite</a> 
          * **PR #10** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/10" target="_blank">Graphite</a> 
            * **PR #11** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/11" target="_blank">Graphite</a> 
              * **PR #12** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/12" target="_blank">Graphite</a> 
              * **PR #18** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/18" target="_blank">Graphite</a> 
                * **PR #20** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/20" target="_blank">Graphite</a> 
                  * **PR #21** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/21" target="_blank">Graphite</a> 
                    * **PR #22** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/22" target="_blank">Graphite</a>  👈
    * **PR #4** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/4" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/22?utm_source=stack-comment).

File: verifier/src/report_body.rs [hunk 1]
```diff
@@ -362,4 +397,27 @@ mod test {
 
         assert_eq!(verifier.verify(&isv_svn).is_some().unwrap_u8(), 1);
     }
+
+    #[test]
+    fn miscellaneous_select_success() {
+        let miscellaneous_select = MiscellaneousSelect::from(REPORT_BODY_SRC.misc_select);
+        let verifier = MiscellaneousSelectVerifier::new(miscellaneous_select.clone());
```
Comment (nick-mobilecoin) [inline verifier/src/report_body.rs:404]: MiscellaneousSelect is a u32, created https://github.com/mobilecoinfoundation/sgx/issues/283 to derive copy until then using `clone()`

**Review by varsha888 — APPROVED** [review]

**Review by jcape — COMMENTED** [review]
This LGTM, but reading through all these, can they be collapsed into either a macro or a blanket impl over generics?

e.g. something like

```rust
/// A marker trait indicating which objects should be verified with ct_eq()
trait VerifiedWithCtEq: ConstantTimeEq {}

impl VerifiedWithCtEq for MiscellaneousSelect {}
impl VerifiedWithCtEq for ConfigId {}
impl VerifiedWithCtEq for ExtProdId {}

pub struct EqualityVerifier<T: VerifiedWithEq + Clone> {
    expected: T
}

impl<T: VerifiedWithCtEq, E: Accessor<T>> for EqualityVerifier<T> {
    type Error = VerificationError;

    fn verify(&self, evidence: &E) -> CtOption<Self::Error> {
        let expected = self.expected.clone();
        let actual = evidence.get();

        let is_eq = expected.ct_eq(actual);
        CtOption::new(
            VerificationError::from((expected, actual)), // maybe use type specialization here?
            !is_eq
        )
    }
}
```

**Review by jcape — APPROVED** [review]

**Comment by nick-mobilecoin** [comment]
bump @varsha888 due to restack

**Comment by nick-mobilecoin** [comment]
> This LGTM, but reading through all these, 
Will work on this as a follow on PR on top of #26 or after all of them are merged, added task to #7

**Review by varsha888 — APPROVED** [review]

**Pull request merged**